File: go/log/log.go
package log

import (
	"fmt"
	"net/url"
	"time"
)
func LogRequest(url *url.URL, path string) {
	now := time.Now()
	if path == "" {
		fmt.Println(fmt.Sprintf("[%s] URL: %s %s" ,
		ColorText(now.Format("2006-01-02 15:04:05"), Black), 
		ColorText(url.Path, Magenta),
		ColorText("Not Found", Red),
	))
	} else {
		fmt.Println(fmt.Sprintf("[%s] Route: %s, URL: %s" ,
		ColorText(now.Format("2006-01-02 15:04:05"), Black), 
		ColorText(path, Blue),
		ColorText(url.Path, Magenta)),
	)
	}

}

File: nodejs/log/log.js
const { Colors } = require("./colors");

const colorText = (text, color = Colors.white) => `\x1b[${color}${text}\x1b[0m`;

const currentTimestamp = `${colorText(new Date().toISOString(), Colors.cyan)}`;

const logRequest = ({ router, url }) => {
  if (!router) {
    return console.log(
      `[${currentTimestamp}] URL: ${colorText(url, Colors.magenta)} ${colorText(
        "Not Found",
        Colors.red
      )}`
    );
  }
  console.log(
    `[${currentTimestamp}] Route: ${colorText(
      router?.path,
      router?.path !== undefined ? Colors.blue : Colors.red
    )} Method: ${colorText(
      router?.method,
      router?.method ? Colors.yellow : Colors.red
    )}, URL: ${colorText(url, Colors.magenta)}`
  );
};

module.exports = {
  logRequest,
};
